url4fdf79ae2c_beckchr_juel_tgz-pJ8-de_odysseus_el_samples_xml_sax_XMELFilterJ8/src/de/odysseus/el/misc/TypeConverterImpl.java

```diff
@@ -0,0 +1,367 @@
+		}
+		if (value instanceof Number) {
+			return Short.valueOf(((Number)value).shortValue());
+		}
+		if (value instanceof String) {
+			try {
+				return Short.valueOf((String)value);
+			} catch (NumberFormatException e) {
+				throw new ELException(LocalMessages.get("error.coerce.value", value, value.getClass(), Short.class));
+			}
+		}
+		if (value instanceof Character) {
+			return Short.valueOf((short)((Character)value).charValue());
+		}
+		throw new ELException(LocalMessages.get("error.coerce.type", value, value.getClass(), Short.class));
+	}
+
+	protected Byte coerceToByte(Object value) {
+		if (value == null || "".equals(value)) {
+			return Byte.valueOf((byte)0);
+		}
+		if (value instanceof Byte) {
+			return (Byte)value;
+		}
+		if (value instanceof Number) {
+			return Byte.valueOf(((Number)value).byteValue());
+		}
+		if (value instanceof String) {
+			try {
+				return Byte.valueOf((String)value);
+			} catch (NumberFormatException e) {
+				throw new ELException(LocalMessages.get("error.coerce.value", value, value.getClass(), Byte.class));
+			}
+		}
+		if (value instanceof Character) {
+			return Byte.valueOf(Short.valueOf((short)((Character)value).charValue()).byteValue());
+		}
+		throw new ELException(LocalMessages.get("error.coerce.type", value, value.getClass(), Byte.class));
+	}
+
+	protected String coerceToString(Object value) {
+		if (value == null) {
+			return "";
+		}
+		if (value instanceof String) {
+			return (String)value;
+		}
+		if (value instanceof Enum<?>) {
+			return ((Enum<?>)value).name();
+		}
+		return value.toString();
+	}
+
+	@SuppressWarnings("unchecked")
+	protected <T extends Enum<T>> T coerceToEnum(Object value, Class<T> type) {
+		if (value == null || "".equals(value)) {
+			return null;
+		}
+		if (type.isInstance(value)) {
+			return (T)value;
+		}
+		if (value instanceof String) {
+			try {
+				return Enum.valueOf(type, (String)value);
+			} catch (IllegalArgumentException e) {
+				throw new ELException(LocalMessages.get("error.coerce.value", value, value.getClass(), type));
+			}
+		}
+		throw new ELException(LocalMessages.get("error.coerce.type", value, value.getClass(), type));
+	}
+
+	protected Object coerceStringToType(String value, Class<?> type) {
+		PropertyEditor editor = PropertyEditorManager.findEditor(type);
+		if (editor == null) {
+			if ("".equals(value)) {
+				return null;
+			}
+			throw new ELException(LocalMessages.get("error.coerce.type", value, String.class, type));
+		} else {
+			if ("".equals(value)) {
+				try {
+					editor.setAsText(value);
+				} catch (IllegalArgumentException e) {
+					return null;
+				}
+			} else {
+				try {
+					editor.setAsText(value);
+				} catch (IllegalArgumentException e) {
+					throw new ELException(LocalMessages.get("error.coerce.value", value, value.getClass(), type));
+				}
+			}
+			return editor.getValue();
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	protected Object coerceToType(Object value, Class<?> type) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.